### ISSUE TITLE
OPINION: Deposit should not be released while username is in use

### ISSUE THREAD

**Issue by fallen-icarus** [ilap/adatag#13]
I like this project! The ethos of this being a **real** DApp not controlled by any entity or keys resonates a lot with me. I've been trying to design my own DApps with a similar ethos.

I didn't know where to say this so apologies if using an Issue is inappropriate. The README describes using a time-lock deposit that gets unlocked after 20 days with the argument that it can prevent abuse of the system. The [adatag website](https://adatag.io/faq) further elaborates on this saying:

> ... This feature is designed to prevent users from buying up all very rare usernames, such as the 1-char ones, for nearly free and then selling them at a very high price. When a user mints an adatag, they must deposit some ADA, proportional to the length of the adatag. This deposit is always claimable only by the minter after some time, i.e., the deadline, which is a protocol parameter. The suggested deadline is 20 days. This ensures that users have a stake in the system and are discouraged from abusing it.

I'm skeptical this will prevent abuse. IMO this deposit doesn't do anything since, after the 20 days, they can get their deposit back while keeping the username. So, in the end, there was no cost to the user above the typical tx fee. All it does is delay how quickly they can purchase the 1-char usernames by 20 days (which isn't that much time). I don't see how this deters users from `buying up all very rare usernames, such as the 1-char ones, for nearly free and then selling them at a very high price`. Plus, after the deposit is reclaimed, the user no longer has any stake in the system.

I think the incentives would be better if either:
- The transaction fee is increased inversely proportionally to the length of the username - shorter usernames should be exponentially more expensive than longer ones. This would be instead of the deposit.
- The deposit is calculated the same as the current design, but it is not unlocked unless the username is burned. The deposit would act like the minUTxOValue requirement but for usernames. This forces the user to have stake in the system for the entire time the username is in use. It also incentivizes users to not keep unused usernames since there is a financial incentive to reclaim the deposit; the larger the deposit, the more users will wish to burn unused usernames.

Perhaps both options can be used together. What are your thoughts on this? Why do you think locking a deposit for 20 days is enough?

**Comment by ilap** [comment]
Hi @fallen-icarus, thanks for your suggestions! In my opinion, none of the proposed solutions would fully prevent the creation of a secondary market. Though, I did choose the simplest one, it's worth reconsidering the best way to prevent abuse.

**Comment by fallen-icarus** [comment]
> prevent the creation of a secondary market

I don't think this should be the goal. IMO the real issue is people buying up usernames just to sell later. In other words, the issue is **purchased usernames go unused**. Having a secondary market is fine since this may be more efficient in some circumstances than just burning and re-minting:

- The secondary market enables users to pay up to guarantee they get it.
- Burning and re-minting creates a race - the first one to re-mint the username gets it.

I believe the price mechanism will be more fair than who can mint the username first since the latter depends on hardware/networking and is susceptible to an MEV attack. 

If the goal is to minimize unused usernames, having an exponential deposit required for the entire time will make users think twice about sitting on usernames. For example, if a really popular username required locking up 500 ADA, a user has a huge incentive to return the username. Even if the username is purchased on the secondary market, if the new owner no longer has a use for it, the 500 ADA incentivizes them to return it. Conversely, if the deposit was reclaimed *while* the username was still in use, there is no longer an incentive to return the unused username.

**Comment by ilap** [comment]
> For example, if a really popular username required locking up 500 ADA, a user has a huge incentive to return the username. Even if the username is purchased on the secondary market, if the new owner no longer has a use for it, the 500 ADA incentivizes them to return it.

Sorry, I misunderstood you, for this this above it makes sense, but for preventing the abuse, as you were referring to it in your first post, I do not think it would be effective, as I would say, again, it would just raise the margin from 0 to the locked deposit. But, who knows.

**Comment by fallen-icarus** [comment]
Hmm, thinking about it more, I suppose there is a chance users still hoard them even if the deposit is high. They would effectively be locking up ada for a few years with the hopes of the usernames potentially being worth more in the future. If their guess about the future is wrong, they can reclaim the ada at very little cost...

Have you thought about how a subscription mechanism could work?
